src/Faker/Generator.php

```diff
@@ -470,7 +470,10 @@ public function randomFloat($nbMaxDecimals = null, $min = 0, $max = null): float
      */
     public function randomNumber($nbDigits = null, $strict = false): int
     {
-        return $this->ext(Extension\NumberExtension::class)->randomNumber((int) $nbDigits, (bool) $strict);
+        return $this->ext(Extension\NumberExtension::class)->randomNumber(
+            $nbDigits !== null ? (int) $nbDigits : null,
```

Comment:
Shame the `(?int)` cast isn't a thing.

Reply:
Haha yeah indeed

Reply:
Wouldn't it make more sense to remove the casting and let the extension handle the null value ? It's already the case for `src/Faker/Core/Number.php`, I'm not sure what other `NumberExtension` there is but it would make a more element solution.

Reply:
I would rather have the core methods strict and handle the input data on the outside.

More since extension is used a lot in other extensions.